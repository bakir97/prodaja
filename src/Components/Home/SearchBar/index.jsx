import React, { Component } from "react";
import SearchBar from "material-ui-search-bar";
const style = {
  searchBar: {
    maxWidth: 400,
    margin: 10,
    marginTop: 0,
    width: "100%"
  }
};
export default class SearchBarComponent extends Component {
  state = {
    search: ""
  };
  render() {
    return (
      <SearchBar
        onCancelSearch={() => this.setState({ search: "" })}
        value={this.state.search}
        onChange={e => this.setState({ search: e })}
        onRequestSearch={() => this.props.search(this.state.search)}
        style={style.searchBar}
      />
    );
  }
}
