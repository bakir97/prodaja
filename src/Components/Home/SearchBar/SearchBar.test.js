import React from "react";
import SearchBar from "./SearchBar";
describe("<SearchBar/>", () => {
  const wrapper = shallow(<SearchBar />);
  it("should render", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("should change state", () => {
    wrapper.simulate("change", "e");
    expect(wrapper.state()).toEqual({ search: "e" });
  });
});
