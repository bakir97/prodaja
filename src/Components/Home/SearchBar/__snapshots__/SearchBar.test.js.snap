// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBar/> should render 1`] = `
<WithStyles(SearchBar)
  onCancelSearch={[Function]}
  onChange={[Function]}
  onRequestSearch={[Function]}
  style={
    Object {
      "margin": 10,
      "marginTop": 0,
      "maxWidth": 400,
      "width": "100%",
    }
  }
  value=""
/>
`;
